bluebutton-data-pipeline-rif-extract/src/main/java/gov/hhs/cms/bluebutton/datapipeline/rif/extract/s3/DataSetTestUtilities.java

```diff
@@ -112,6 +116,16 @@ public static PutObjectRequest createPutRequest(Bucket bucket, DataSetManifest m
 			ObjectMetadata objectMetadata = new ObjectMetadata();
 			objectMetadata.setContentLength(objectContentLength);
 
+			// create md5chksum on file to be uploaded
+			Path fileToUploadPath = null;
+			try {
+				fileToUploadPath = Paths.get(objectContentsUrl.toURI());
+			} catch (URISyntaxException e) {
+				e.printStackTrace();
+			}
+			objectMetadata.addUserMetadata("md5chksum",
+					ManifestEntryDownloadTask.computeMD5ChkSum(fileToUploadPath));
+
```

Comment:
Can you ask Mike if he's uploading the manifests with a checksum, too? If so, you should probably verify those as well. (Though you could split that off into a separate JIRA ticket if you'd like.)

Reply:
Yes, the manifest files are being uploaded with a checksum also.   I can create a ticket.